tests/batch_size_optimizer/test_simulator.py

```diff
@@ -0,0 +1,161 @@
+import asyncio
+import logging
+import re
+from typing import AsyncIterator
+import pytest
+from examples.trace_driven.run_single import read_trace
+from fastapi.testclient import TestClient
+from typing import Literal
+from pytest_mock import MockerFixture
+from sqlalchemy.ext.asyncio.session import AsyncSession
+from tests.batch_size_optimizer.simulate_with_server import SimulatorWithServer
+from zeus.job import Job
+from zeus.optimizer.batch_size.server.database.db_connection import (
+    DatabaseSessionManager,
+    get_db_session,
+)
+from zeus.optimizer.batch_size.server.database.schema import Base
+from zeus.optimizer.batch_size.server.router import app
+from zeus.policy.optimizer import JITPowerLimitOptimizer, PruningGTSBatchSizeOptimizer
+from zeus.simulate import Simulator
+
+
+config = {
+    "gpu": "v100",
+    "eta_knob": 0.5,
+    "beta_knob": 2.0,
+    "seed": 1,
+    "dataset": "librispeech",
+    "model": "deepspeech2",
+    "optimizer": "adamw",
+    "target_metric": 40.0,
+    "max_epochs": 16,
+    "b_0": 192,  # default_bs
+    "num_recurrence": None,
+}
```

Comment:
Can you port more test cases? We discussed this maybe two months ago, when I asked you to run `examples/trace_driven` and port them into test cases.

Reply:
You mean a different configuration?

Reply:
Yeah not just deepspeech2 on V100, but more configurations